src/command_modules/azure-cli-resource/azure/cli/command_modules/resource/custom.py

```diff
@@ -64,6 +73,110 @@ def create_resource_group(rg_name, location, tags=None):
     )
     return rcf.resource_groups.create_or_update(rg_name, parameters)
 
+def create_appliance(resource_group_name, appliance_name, managed_rg_id, location, kind, appliance_definition_id=None, plan_name=None, plan_publisher=None, plan_product=None, plan_version=None, tags=None, parameters=None):
+    ''' Create a new appliance.
+    :param str resource_group_name:the desired resource group name
+    :param str appliance_name:the appliance name
+    :param str kind:the appliance kind. can be marketplace or servicecatalog
+    :param str plan_name:the appliance package plan name
+    :param str plan_publisher:the appliance package plan publisher
+    :param str plan_product:the appliance package plan product
+    :param str plan_version:the appliance package plan version
+    :param str tags:tags in 'a=b c' format
+    '''
+    racf = _resource_appliances_client_factory()
+    applianceProperties = ApplianceProperties(managed_rg_id)
+    appliance = Appliance(
+        location=location,
+        properties=applianceProperties,
+        kind=kind,
+        tags=tags
+    )
+
+    if kind.lower() == 'servicecatalog':
+        if appliance_definition_id:
+            applianceProperties = ApplianceProperties(managed_rg_id, appliance_definition_id)
+        else:
+            raise CLIError('--appliance-definition-id is required if kind is ServiceCatalog')
+    elif kind.lower() == 'marketplace':
+        if plan_name is None and plan_product is None and plan_publisher is None and plan_version is None:
+            raise CLIError('--plan-name, --plan-product, --plan-publisher and --plan-version are all required if kind is ServiceCatalog')
+        else:
+            appliancePlan = Plan(plan_name, plan_publisher, plan_product, plan_version)
+            appliance.plan = appliancePlan
+
+    applianceParameters = None
+
+    if parameters:
+        if os.path.exists(parameters):
+            applianceParameters = get_file_json(parameters)
+        else:
+            applianceParameters = shell_safe_json_parse(parameters)
+
+    applianceProperties.parameters = applianceParameters
+
+    return racf.appliances.create_or_update(resource_group_name, appliance_name, appliance)
+
+def show_appliance(resource_group_name=None, appliance_name=None, appliance_id=None):
```

Comment:
It is in the params file. We don't necessarily add param description for all params in custom file